camelot/view/action_steps/item_view.py

```diff
@@ -306,57 +288,21 @@ def render(cls, gui_context, step, engine):
         list(new_model.add_columns(step['columns']))
         new_model.set_value(step['proxy_route'])
 
-        # create QML item
-        view = create_qml_item(
-            #QtCore.QUrl("qrc:/Vortex/qml/common/TablePage.qml"),
-            QtCore.QUrl("qrc:/Vortex/TableView/TablePage.qml"),
-            { 'model': new_model, 'headerModel': header_model },
-            engine
-        )
-        new_model.setParent(view)
-        header_model.setParent(view)
-
-        # load JSON data into C++ backend
-        view.fromJson(json.dumps(step))
+        context_id = qml_action_step(list_gui_context, 'OpenTableView',
+                serialized_step, { 'model': new_model })
 
-        table = view.findChild(QtCore.QObject, "qml_table")
+        window = get_qml_window()
+        view = window.findChild(QtQuick.QQuickItem, 'qml_table_view_{}'.format(context_id))
+        table = view.findChild(QtQuick.QQuickItem, 'qml_table')
```

Comment:
zijn we hier niet gewoon 'lucky' als dat die item reeds beschikbaar is

moeten we nt op een of andere manier wachten tot het item er is ?  we kunnen dat item mss ook reparenten aan de backend van die table view ?

Reply:
Nee. Het zijn geen netwerk URL, het aanmaken van het QQmlComponent in C++ is dus synchroon. Het aanmaken van de instance met QQmlComponent::create() is ook synchroon en voert volgende stappen uit [1]:

1. The object hierarchy is created, and constant values are assigned.
2. Property bindings are evaluated for the first time.
3. If applicable, QQmlParserStatus::componentComplete() is called on objects.

[1] https://doc.qt.io/qt-5/qqmlcomponent.html#beginCreate